mlflow/pyfunc/dbconnect_artifact_cache.py

```diff
@@ -0,0 +1,135 @@
+import json
+import os
+import subprocess
+from mlflow.utils.databricks_utils import is_in_databricks_runtime
+from mlflow.utils.file_utils import get_or_create_tmp_dir
+
+
+_global_dbconnect_artifact_cache = None
+
+_CACHE_MAP_FILE_NAME = "db_connect_artifact_cache.json"
+
+
+class DBConnectArtifactCache:
+    """
+    This class is designed for managing Databricks Connect artifacts cache.
+    Note it doesn't support OSS Spark Connect.
+
+    You can use this class in the following environment:
+     - Databricks shared cluster python notebook REPL
+     - Databricks Serverless python notebook REPL
+     - Databricks connect client python REPL that connects to remote Databricks Serverless
+     - Databricks connect client python REPL that connects to remote Databricks shared cluster
+
+    .. code-block:: python
+        :caption: Example
+
+        # client side code
+        db_artifact_cache = DBConnectArtifactCache.get_or_create()
+        db_artifact_cache.add_artifact_archive("archive1", "/tmp/archive1.tar.gz")
+
+        @pandas_udf(...)
+        def my_udf(...):
+           # we can get the unpacked archive files in `archive1_unpacked_dir`
+           archive1_unpacked_dir = db_artifact_cache.get("archive1")
+    """
+
+    @classmethod
```

Comment:
```suggestion
    @staticmethod
```

because `cls` isn't used.